### ISSUE TITLE
Any plans on a mj-include alternative?

### ISSUE THREAD

**Issue by dimitarnestorov** [SparkPost/heml#6]
[mj-include](https://mjml.io/documentation/#mj-include) tag is my deal breaker, a way to include headers and footers is a must.
Is something similar or even better planned for the future?

**Comment by avigoldman** [comment]
Yeah, definitely. Are there any improvements you'd want to make on top of the `mj-include` tag that I can try for in HEML? Do you expect it to work with both remote and local files?

**Comment by dimitarnestorov** [comment]
My situation is a separate repository for the emails I send and what I notice is that the `mj-include` path is relative to the folder in which `mjml` is ran. I don't know about other people but since I use webpack and node all the time every file path has to be relative to the file itself, and I will assume that most devs think that way.

Example:
![image](https://user-images.githubusercontent.com/8790386/32131192-50889c62-bbb0-11e7-9146-b33cb2c0f6a1.png)

You can see that the paths of the includes are relative to the root of the project. Doesn't feel right.

How I think it should be:
![image](https://user-images.githubusercontent.com/8790386/32131209-d3c8eb22-bbb0-11e7-9b35-d6726d9b12c8.png)

Also it looks like it supports inclusion of styles now that I look at it.

And as per your question for remote files, everything is local for me.

At the end of the day if HEML has basic markup and style include support I guess I will be switching. Love the fact that you focus on responsive, it's 2017 after all. Mailjet seem to be focused on email client support.

**Comment by avigoldman** [comment]
Cool, this is great feedback! I'll make sure to use when I start building the `include` element out. I'll share it here to get your feedback too!

**Comment by thommeredith** [comment]
Any update on this?

**Comment by pateketrueke** [comment]
I also want this landed soon, effectively using pipelines or pre-processors just work out-of-the-box, e.g. `pug + heml + mustache` to pre-compile all my templates programmatically.

While this is working pretty fine, I lost the preview feature from the CLI, not critical yet...

Another suggestion would be support a kind-of pre-processing support, so we can pipe anything before HEML does its job. 

Thank you!

**Comment by lkraav** [comment]
I think there must be something better out, where all the crowds are gathering and putting in work. This project unfortunately seems to have lost momentum.

**Comment by pateketrueke** [comment]
Well, on my side I'll see if I can implement this feature by hand (or pre-processing), and then open a PR.

If not, I'll be doing research for something I'm heavily using at my current job.
